avocado/core/future/settings.py

```diff
@@ -0,0 +1,516 @@
+                                     help_msg='this is just a test')
+
+        This will register a 'foo.bar' namespace inside Avocado internals
+        settings. And this could be now, be changed by the users or system
+        configuration option:
+
+           [foo]
+           bar = a different message replacing 'hello'
+
+        If you would like to provide also the flexibility to the user change
+        the values via command-line, you sould pass the other arguments.
+
+        Arguments
+        ---------
+        section : str
+            The configuration file section that your option should be present.
+            You can specify subsections with dots. i.e: run.output.json
+
+        key : str
+            What is the key name of your option inside that section.
+
+        default : typeof(key_type)
+            What is the default value of your option if the key is not present
+            inside the section on all configuration files.
+
+        help_msg : str
+            The help message that will be displayed at command-line (-h) and
+            configuration file template.
+
+        key_type : any method
+            What is the type of your option? Currently supported: int, list,
+            str or a custom method. Default is `str`.
+
+        parser : argparser parser
+            Since that you would like to have a command-line option, you should
+            specify what is the parser or parser group that we should add this
+            option.
+
+        positional_arg : bool
+            If this option is an positional argument or not. Default is
+            `False`.
+
+        short_arg : str
+            A short option for the command-line. i.e: `-d` for debug.
+
+        long_arg: : str
+            A long option for the command-line. i.e: `--debug` for debug.
+
+        choices : typle
+            If you would like to limit the option to a few choices. i.e:
+            ('foo', 'bar')
+
+        nargs : int or str
+            The number of command-line arguments that should be consumed. Could
+            be a int, '?', '*' or '+'. For more information visit the argparser
+            documentation.
+
+        metavar : str
+            String presenting available sub-commands in help, if None we will
+            use the section+key as metavar.
+
+        required : bool
+            If this is a required option or not when on command-line. Default
+            is False.
+
+        action :
+            The basic type of action to be taken when this argument is
+            encountered at the command line. For more information visit the
+            argparser documentation.
+
+        .. note:: Most of the arguments here (like parser, positional_arg,
+                  short_arg, long_arg, choices, nargs, metavar, required and
+                  action) are only necessary if you would like to add a
+                  command-line option.
+        """
```

Comment:
This is a great docstring. Do you think it could be improved by also referring to the respective `argparse` arguments?

Reply:
Not sure if I got your comment.

IMO, a perfect scenario would be to abstract all the internal details from the API user. However, today we are still using `choices`, `nargs`, `metavar`, `required` and `action` that are required by the argparser and they are there.

On `parser` argument also I noted the type "argparser parser".

So, I'm not sure what to do here. @clebergnu could you eli5? :)